agent/consul/server.go

```diff
@@ -939,7 +939,22 @@ func isV1CatalogRequest(rpcName string) bool {
 }
 
 func (s *Server) registerControllers(deps Deps, proxyUpdater ProxyUpdater) error {
-	hcpctl.RegisterControllers(s.controllerManager)
+	// TODO(CC-6389): Remove once resource-apis is no longer considered experimental and is supported by HCP
+	overrideResourceApisEnabledCheckRaw := os.Getenv("CONSUL_OVERRIDE_HCP_RESOURCE_APIS_CHECK")
+	overrideResourceApisEnabledCheck := false
+	if overrideResourceApisEnabledCheckRaw != "" {
```

Comment:
nit: I think `strconv.ParseBool(overrideResourceApisEnabledCheckRaw)` below will return false if the value is set to an empty string. But not really a big deal either way!

Reply:
It does return false, but also an error in that case:
```
func ParseBool(str string) (bool, error) {
	switch str {
	case "1", "t", "T", "true", "TRUE", "True":
		return true, nil
	case "0", "f", "F", "false", "FALSE", "False":
		return false, nil
	}
	return false, syntaxError("ParseBool", str)
}
```
Seemed a bit cleaner to me to not attempt to parse if the env var is empty.